pkg/build/graph/analysis/bc.go

```diff
@@ -98,3 +102,71 @@ func FindCircularBuilds(g osgraph.Graph) []osgraph.Marker {
 
 	return markers
 }
+
+func FindPendingTags(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+ist:
+	for _, uncastIstNode := range g.NodesByKind(imagegraph.ImageStreamTagNodeKind) {
+		istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode)
+		if bcNode, points := buildPointsToTag(g, uncastIstNode); points && !istNode.Found() {
+			// A build config points to the non existent tag but the latest build has failed.
+			// Since the tag isn't yet populated, there hasn't been any successful build so far.
+			if buildNode, failed := latestBuildFailed(g, bcNode); failed {
+				markers = append(markers, osgraph.Marker{
+					Node:         buildNode,
+					RelatedNodes: []graph.Node{uncastIstNode, bcNode, buildNode},
+
+					Severity:   osgraph.ErrorSeverity,
+					Key:        LatestBuildFailedErr,
+					Message:    fmt.Sprintf("%s has failed.", buildNode.(*buildgraph.BuildNode).ResourceString()),
+					Suggestion: osgraph.Suggestion(fmt.Sprintf("Inspect the build failure with 'oc logs %s'", buildNode.(*buildgraph.BuildNode).ResourceString())),
+				})
+				continue ist
+			}
+
+			// A build config points to the non existent tag and maybe no current build exists.
+			marker := osgraph.Marker{
+				Node:         bcNode,
+				RelatedNodes: []graph.Node{uncastIstNode},
+
+				Severity: osgraph.WarningSeverity,
+				Key:      TagNotAvailableWarning,
+				Message:  fmt.Sprintf("%s needs to be imported or created by a build.", istNode.ResourceString()),
+			}
+			if !hasBuild(g, bcNode) {
+				marker.Suggestion = osgraph.Suggestion(fmt.Sprintf("oc start-build %s", bcNode.(*buildgraph.BuildConfigNode).ResourceString()))
```

Comment:
> I have a running build there. Will post two more (one with this and one with a failed build).

Hmmm... I hadn't considered that.  Think we should skip the warning if a build is in progress?